File: Java/src/session2/td/heritageMultiple/SuccDecimal.java
package session2.td.heritageMultiple;

import session1.td.FabriqueNaturels;
import session1.td.Nat;


public class SuccDecimal extends EtatSucc implements AlgebreNatDecimal {
    public static final FabriqueNaturels<Nat> FAB = new SuccDecimal(new ZeroDecimal());

    public SuccDecimal(Nat predecesseur) {
        super(predecesseur);
    }



    @Override
    public Nat creerZero() {
        return new ZeroDecimal();
    }

    @Override
    public Nat creerSuccesseur(Nat predecesseur) {
        return new SuccDecimal(predecesseur);
    }

    @Override
    public String toString() {
        return ""+this.val();
    }

    @Override
    public boolean equals(Object obj) {
        if (!(obj instanceof Nat))
            return false;
        Nat x = (Nat) obj;
        return this.val() == x.val();
    }

}

File: TypeScript/src/session2.td/heritageMultiple/SuccDecimal.ts
import {Nat} from "../../session1.td/Nat";
import {FabriqueNaturels} from "../../session1.td/FabriqueNaturels";
import {EtatSucc} from "./EtatSucc";
import {AlgebreNatDecimal, AlgebreNatDecimal_default} from "./AlgebreNatDecimal";
import {ZeroDecimal} from "./ZeroDecimal";
import {heriterInterfacesConcretes} from "./heritageInterfacesconcretes";

export class SuccDecimal extends EtatSucc implements AlgebreNatDecimal {
    public static FAB : FabriqueNaturels<Nat> = new SuccDecimal(new ZeroDecimal());

    public constructor(predecesseur : Nat) {
        super(predecesseur);
    }

    public creerZero() : Nat {
        return new ZeroDecimal();
    }

    public creerSuccesseur(predecesseur : Nat) : Nat {
        return new SuccDecimal(predecesseur);
    }

    public toString() : string {
        return String(this.val());
    }

    public equals(obj : Nat) : boolean {
        return this.val() == obj.val();
    }

    div(x: Nat): Nat {
        return undefined;
    }

    modulo(x: Nat): Nat {
        return undefined;
    }

    produit(x: Nat): Nat {
        return undefined;
    }

    somme(x: Nat): Nat {
        return undefined;
    }

    un(): Nat {
        return undefined;
    }

    zero(): Nat {
        return undefined;
    }

}


heriterInterfacesConcretes(SuccDecimal, [EtatSucc, AlgebreNatDecimal_default]);